framework/db/transaction_manager.py

```diff
@@ -438,6 +441,43 @@ def GetByIDAsDict(self, trans_id, target_id=None):
         return self.DeriveTransactionDict(transaction_obj, include_raw_data=True)
 
     @target_required
+    def GetHrtResponse(self, filter_data, trans_id, target_id=None):
+        transaction_obj = self.db.session.query(models.Transaction).filter_by(target_id=target_id, id=trans_id).first()
+
+        # Data validation
+        languages = ['bash']  # Default script language is set to bash.
+        if filter_data.get('language'):
+            languages = map(lambda x: x.strip(), filter_data['language'][0].split(','))
```

Comment:
@saganshul @DePierre I don't know why we do `dict(self.request.arguments)`. According to the Tornado docs, `self.request.arguments` is a dictionary. Each key of that dictionary has a value which is a list.
Now `filter_data['language']` will give us a list. Hence the index `[0]`.